client/src/client/mod.rs

```diff
@@ -0,0 +1,35 @@
+use hyper;
+use hyper_tls::HttpsConnector;
+use tokio_core::reactor::Handle;
+use error::{Result, Error};
+
+/// A client that can issue requests to a horizon api.
+#[derive(Debug)]
+pub struct Client {
+    inner: hyper::Client<HttpsConnector<hyper::client::HttpConnector>>,
+    uri: hyper::Uri,
+}
+
+impl Client {
+    /// Constructs a new stellar client.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// # extern crate tokio_core;
+    /// # extern crate stellar_client;
+    /// # fn main() {
+    /// use tokio_core::reactor::Core;
+    /// use stellar_client::Client;
+    /// let core = Core::new().unwrap();
+    /// let client = Client::new("https://horizon-testnet.stellar.org", &core.handle());
+    /// # }
+    /// ```
+    pub fn new(uri: &str, handle: &Handle) -> Result<Self> {
```

Comment:
URI feels more like a configuration to me than a parameter, but I'm happy to come back to that later.

Reply:
I'm not sure what you mean?

In order to create an instance of the client you need to hand it a uri in some form. This could be through a builder pattern but since it's just 2 params at the moment I felt a param made more sense. It's how you "configure" the client.